commit 1dbe6f76677c26096518998fdc72dab771a98913
Author: Michael Paquier <michael@paquier.xyz>
Date:   Wed Jul 16 11:59:22 2025 +0900

    Refactor non-supported compression error message in toast_compression.c
    
    This code used a NO_LZ4_SUPPORT() macro to issue an error in the code
    paths where LZ4 [de]compression is attempted but the build does not
    support it.  This commit refactors the code to use a more flexible error
    message so as it can be used for other compression methods, where the
    method is given in input of macro.
    
    Extracted from a larger patch by the same author.
    
    Author: Nikhil Kumar Veldanda <veldanda.nikhilkumar17@gmail.com>
    Reviewed-by: Kirill Reshke <reshkekirill@gmail.com>
    Discussion: https://postgr.es/m/CAFAfj_HX84EK4hyRYw50AOHOcdVi-+FFwAAPo7JHx4aShCvunQ@mail.gmail.com

diff --git a/src/backend/access/common/toast_compression.c b/src/backend/access/common/toast_compression.c
index 21f2f4af97e..926f1e4008a 100644
--- a/src/backend/access/common/toast_compression.c
+++ b/src/backend/access/common/toast_compression.c
@@ -23,15 +23,15 @@
 #include "varatt.h"
 
 /* GUC */
 int			default_toast_compression = TOAST_PGLZ_COMPRESSION;
 
-#define NO_LZ4_SUPPORT() \
+#define NO_COMPRESSION_SUPPORT(method) \
 	ereport(ERROR, \
 			(errcode(ERRCODE_FEATURE_NOT_SUPPORTED), \
-			 errmsg("compression method lz4 not supported"), \
-			 errdetail("This functionality requires the server to be built with lz4 support.")))
+			 errmsg("compression method %s not supported", method), \
+			 errdetail("This functionality requires the server to be built with %s support.", method)))
 
 /*
  * Compress a varlena using PGLZ.
  *
  * Returns the compressed varlena, or NULL if compression fails.
@@ -137,11 +137,11 @@ pglz_decompress_datum_slice(const struct varlena *value,
  */
 struct varlena *
 lz4_compress_datum(const struct varlena *value)
 {
 #ifndef USE_LZ4
-	NO_LZ4_SUPPORT();
+	NO_COMPRESSION_SUPPORT("lz4");
 	return NULL;				/* keep compiler quiet */
 #else
 	int32		valsize;
 	int32		len;
 	int32		max_size;
@@ -180,11 +180,11 @@ lz4_compress_datum(const struct varlena *value)
  */
 struct varlena *
 lz4_decompress_datum(const struct varlena *value)
 {
 #ifndef USE_LZ4
-	NO_LZ4_SUPPORT();
+	NO_COMPRESSION_SUPPORT("lz4");
 	return NULL;				/* keep compiler quiet */
 #else
 	int32		rawsize;
 	struct varlena *result;
 
@@ -213,11 +213,11 @@ lz4_decompress_datum(const struct varlena *value)
  */
 struct varlena *
 lz4_decompress_datum_slice(const struct varlena *value, int32 slicelength)
 {
 #ifndef USE_LZ4
-	NO_LZ4_SUPPORT();
+	NO_COMPRESSION_SUPPORT("lz4");
 	return NULL;				/* keep compiler quiet */
 #else
 	int32		rawsize;
 	struct varlena *result;
 
@@ -287,11 +287,11 @@ CompressionNameToMethod(const char *compression)
 	if (strcmp(compression, "pglz") == 0)
 		return TOAST_PGLZ_COMPRESSION;
 	else if (strcmp(compression, "lz4") == 0)
 	{
 #ifndef USE_LZ4
-		NO_LZ4_SUPPORT();
+		NO_COMPRESSION_SUPPORT("lz4");
 #endif
 		return TOAST_LZ4_COMPRESSION;
 	}
 
 	return InvalidCompressionMethod;
